cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/HttpHelper.java

```diff
@@ -0,0 +1,50 @@
+package zone.cogni.libs.sparqlservice.impl;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class HttpHelper {
+
+  public static HttpClientContext createAuthenticationHttpContext(String user, String password) {
```

Comment:
name of method does not really mention it is doing basic credential. not sure if there are "more secure" ways of creating a `HttpClientContext`

Reply:
It creates an HttpClientContext that can authenticate with given username/password.
The authentication "system" will be handled by the httpClient lib. For example server can ask to authenticate via digest or basic auth. The lib will then kindof use the context to find the username/password.